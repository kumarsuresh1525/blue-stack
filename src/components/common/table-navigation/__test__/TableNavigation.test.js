import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import TableNavigation from '../index';

describe("Table navigation test", () => {
  const defaultProps = {
    loadData: jest.fn()
  }
  test("should render table navigation component", () => {
    const {container} = render(<TableNavigation {...defaultProps}/>);
    expect(container).toMatchSnapshot();
  });

  it('it should handle tab change', () => {
    const {getAllByTestId, container} = render(<TableNavigation {...defaultProps} />);
    const tab = getAllByTestId('tab-navigation');
    fireEvent.click(tab[0]);
    expect(container.querySelector('.active').textContent).toBe('Upcoming Campaigns');
    fireEvent.click(tab[1]);
    expect(container.querySelector('.active').textContent).toBe('Live Campaigns');
    fireEvent.click(tab[2]);
    expect(container.querySelector('.active').textContent).toBe('Past Campaigns');
  });
});