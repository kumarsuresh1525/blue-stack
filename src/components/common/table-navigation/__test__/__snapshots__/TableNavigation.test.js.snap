// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table navigation test should render table navigation component 1`] = `
<div>
  <div
    class="table-navigation row"
  >
    <div
      class="col-12 col-lg-3"
    >
      <div
        class="active"
        data-testid="tab-navigation"
      >
        Upcoming Campaigns
      </div>
    </div>
    <div
      class="col-12 col-lg-3"
    >
      <div
        class=""
        data-testid="tab-navigation"
      >
        Live Campaigns
      </div>
    </div>
    <div
      class="col-12 col-lg-3"
    >
      <div
        class=""
        data-testid="tab-navigation"
      >
        Past Campaigns
      </div>
    </div>
  </div>
</div>
`;
